File: js/Vec3.js
 class Vec3 {
     //三维向量
     constructor(x, y, z) {
         if (z != undefined) {
             this.x = x
             this.y = y
             this.z = z
         } else if (x != undefined) {
             this.x = x.x
             this.y = x.y
             this.z = x.z
         } else {
             this.x = 0
             this.y = 0
             this.z = 0
         }
     }
     //重载 == 运算符
     eq(other) {
         return this.x == other.x && this.y == other.y && this.z == other.z
     }
     //重载 - 运算符
     minus(other) {
         return new Vec3(this.x - other.x, this.y - other.y, this.z - other.z)
     }
     //重载 + 运算符
     plus(other) {
         if (typeof (other) == "number") {
             return new Vec3(this.x + other, this.y + other, this.z + other)
         } else {
             return new Vec3(this.x + other.x, this.y + other.y, this.z + other.z)
         }
     }
     //重载 * 运算符
     mtimes(other) {
         if (typeof (other) == "number") {
             return new Vec3(this.x * other, this.y * other, this.z * other)
         } else {
             return new Vec3(this.x * other.x, this.y * other.y, this.z * other.z)
         }
     }
     //重载 ^ 运算符
     mpower(n) {
         return new Vec3(Math.pow(this.x, n), Math.pow(this.y, n), Math.pow(this.z, n))
     }
 }
 module.exports = Vec3

File: matlab/Vec3.m
classdef Vec3
    %Vec3 三维向量
    
    properties
        x = 0
        y = 0
        z = 0
    end
    
    methods
        function obj = Vec3(x,y,z)
            %Vec3 构造此类的实例
            if  nargin == 3
                obj.x = x;
                obj.y = y;
                obj.z = z;
            elseif nargin == 1
                obj.x = x(1);
                obj.y = x(2);
                obj.z = x(3);
            else
                obj.x = 0;
                obj.y = 0;
                obj.z = 0;
            end
        end
        
        function b = eq(obj,other)
            %eq 重载 == 运算符
            b = obj.x == other.x && obj.y == other.y && obj.z == other.z;
        end
        function b = minus(obj,other)
            %minus 重载 - 运算符
            b = Vec3(obj.x - other.x,obj.y - other.y,obj.z - other.z);
        end
        function b = plus(obj,other)
            %plus 重载 + 运算符
            if isa(other, 'double')
                b = Vec3(obj.x + other,obj.y + other,obj.z + other);
            elseif isa(obj, 'double')
                b = Vec3(obj + other.x,obj + other.y,obj + other.z);
            else
                b = Vec3(obj.x + other.x,obj.y + other.y,obj.z + other.z);
            end
        end
        function b = mtimes(obj,other)
            %mtimes 重载 * 运算符
            if isa(other, 'double')
                b = Vec3(obj.x * other,obj.y * other,obj.z * other);
            elseif isa(obj, 'double')
                b = Vec3(obj * other.x,obj * other.y,obj * other.z);
            elseif isa(other, 'Vec3')
                b = Vec3(obj.x * other.x,obj.y * other.y,obj.z * other.z);
            end
        end
        function b = mpower(obj,n)
            %mpower 重载 ^ 运算符
            b = Vec3(power(obj.x,n),power(obj.y,n),power(obj.z,n));
        end
    end
end

